src/ocf_core.c

```diff
@@ -251,25 +243,14 @@ void ocf_core_submit_io_mode(struct ocf_io *io, ocf_cache_mode_t cache_mode)
 		return;
 	}
 
+	req->part_id = ocf_part_class2id(cache, io->io_class);
+	req->core = core;
+	req->complete = ocf_req_complete;
+
 	/* TODO: instead of casting ocf_cache_mode_t to ocf_req_cache_mode_t
 	   we can resolve IO interface here and get rid of the latter. */
-	req_cache_mode = cache_mode;
-
 	if (cache_mode == ocf_cache_mode_none)
```

Comment:
Move this check to ocf_resolve_effective_cache_mode  and call ocf_resolve_effective_cache_mode unconditionally.

Pls check whether the same change applies to submit_fast.